source/security/authentication.txt

```diff
@@ -0,0 +1,401 @@
+
+First, the {+php-library+} checks whether you passed AWS credentials
+to the ``MongoDB\Client`` constructor, either as an options parameter or as part of the
+connection URI. To pass your credentials to ``MongoDB\Client``,
+set the following connection options:
+
+- ``username``: The AWS IAM access key ID to authenticate. Percent-encode this value
+  before including it in a connection URI.
+- ``password``: The AWS IAM secret access key. Percent-encode this value before including
+  it in a connection URI.
+- ``authMechanism``: Set to ``'MONGODB-AWS'``.
+
+You can set these options in two ways: by passing an options array to the
+``MongoDB\Client`` constructor or through parameters in your connection URI.
+
+.. tabs::
+
+   .. tab:: MongoDB\\Client
+      :tabid: Client
+
+      .. code-block:: php
+
+         $options = [
+             'username' => '<AWS IAM access key ID>',
+             'password' => '<AWS IAM secret access key>',
+             'authMechanism' => 'MONGODB-AWS',
+         ];
+
+         $client = new Client(
+            'mongodb://<hostname>:<port>',
+            $options,
+         );
+
+   .. tab:: Connection URI
+      :tabid: connectionstring
+
+      .. code-block:: php
+
+         $uri = 'mongodb://<AWS IAM access key ID>:<AWS IAM secret access key>@<hostname>:<port>/?authMechanism=MONGODB-AWS';
+         $client = new Client($uri);
+
+.. _php-mongodb-aws-env-vars:
+
+Environment Variables
+~~~~~~~~~~~~~~~~~~~~~
+
+If you don't provide a username and password when you construct your ``MongoDB\Client``
+object, the {+php-library+} tries to retrieve AWS credentials from the following
+environment variables:
+
+- ``AWS_ACCESS_KEY_ID``
+- ``AWS_SECRET_ACCESS_KEY``
+- ``AWS_SESSION_TOKEN``
+
+To use these environment variables to authenticate your application, first set them to the
+AWS IAM values needed for authentication, as shown in the following code
+example:
+
+.. code-block:: sh
+
+   export AWS_ACCESS_KEY_ID=<AWS IAM access key ID>
+   export AWS_SECRET_ACCESS_KEY=<AWS IAM secret access key>
+   export AWS_SESSION_TOKEN=<AWS session token>
+
+.. important::
+  
+   Don't percent-encode the values in these environment variables.
+
+After you set these environment variables, set the ``authMechanism``
+connection option to ``'MONGODB-AWS'``.
+
+.. _php-mongodb-aws-env-example:
+
+Example
+```````
+
+The following example sets the ``authMechanism`` connection option. 
+You can set this option in two ways: by passing an options array to the
+``MongoDB\Client`` constructor or through a parameter in your connection URI.
+
+.. tabs::
+
+   .. tab:: MongoDB\\Client
+      :tabid: Client
+
+      .. code-block:: php
+
+         $client = new Client(
+            'mongodb://<hostname>:<port>',
+            ['authMechanism' => 'MONGODB-AWS'],
+         );
+
+   .. tab:: Connection URI
+      :tabid: connectionstring
+
+      .. code-block:: php
+
+         $uri = 'mongodb://<hostname>:<port>/?&authMechanism=MONGODB-AWS';
```

Comment:
```suggestion
         $uri = 'mongodb://<hostname>:<port>/?authMechanism=MONGODB-AWS';
```

"&" is only a delimiter for sequential URI options. This would have been a parse error.